python/tank/util/qt_importer.py

```diff
@@ -55,35 +55,35 @@ def QtCore(self):
         """
         :returns: QtCore module, if available.
         """
-        return self._modules["QtCore"] if self._modules else None
+        return self._modules.get("QtCore")
```

Comment:
The previous syntax implies `self._modules` may not be set when called so you may need this instead:
```suggestion
        return self._modules.get("QtCore") if self._modules else None
```

This covers QtCore or the module not being present in the dict and the dict itself not being set.

Reply:
If the _modules dict is initialised to an empty dict, get should still be safe, i.e. 

```
>>> d = {}
>>> print(d.get('foobar'))
None
```

If the issue is covering a scenario where _modules is not an attr of the object at all, I don't think we'd need to do that as it's in the constructor?

Reply:
I would prefer the suggestion by @halil3d above, instead of initializing `self._modules` in the constructor to an empty dictionary, because for a couple of minor reasons:

1. If `_modules` is None, this clearly indicates that the import failed for the requested Qt interface. An empty dict may suggest a different error during the import.
2. The `modules` property returns the the value of `_modules` - so we would be changing the return value from `None` to `{}` if the import failed. Which means if callers are checking specifically for `None`, this would break their code.